lib/metadata/Metadata.js

```diff
@@ -12,17 +13,32 @@ class Metadata {
 
     this.ns = namespace(this.factory)
 
+    this.tableSchemas = []
     this.parse()
   }
 
   parse () {
     this.parseDialect()
 
-    this.tableSchemas = [new TableSchema(this.dataset, {
-      baseIRI: this.baseIRI,
-      factory: this.factory,
-      timezone: this.timezone
-    })]
+    if (!this.dataset) {
+      return
+    }
+
+    const tableSchemaNodes = RdfUtils.findNodes(this.dataset, null, this.ns.tableSchema)
+
+    this.tableSchemas = tableSchemaNodes.reduce((arr, tableSchema) => {
+      const table = this.dataset.match(null, this.ns.tableSchema, tableSchema).toArray().shift().subject
+      const tableGroup = this.dataset.match(null, this.ns.table, table).toArray().shift().subject
+
+      return [...arr, new TableSchema(this.dataset, {
+        baseIRI: this.baseIRI,
+        factory: this.factory,
+        timezone: this.timezone,
+        root: tableSchema,
+        table,
+        tableGroup
+      })]
+    }, [])
```

Comment:
This is predominantly why the test fails in `ObjectParserTransform.test.js`

Previously the `root` (table schema) was never set. This PR gathers actual nodes for the schema/table/group and uses them to initialise the `TableSchema` object

Reply:
I propose we remove that test altogether. Is feels quite synthetic, awkward to manage and the e2e test cases cover  all the paths